.github/workflows/ci.yml

```diff
@@ -23,7 +23,7 @@ jobs:
 
       - name: Restore virtualenv from cache
         id: cache
-        uses: actions/cache@v1
+        uses: actions/cache@v3
```

Comment:
Could you please also bump other actions version?

- actions/checkout@v2 -> actions/checkout@v4
- actions/setup-python@v2 -> actions/setup-python@v5

Could you please move these changes to a separate commit with the following comment: `chore(ci): bump actions version`